Assets/MixedRealityToolkit.Tests/EditModeTests/SpatialAwareness/SpatialAwarenessSystemTests.cs

```diff
@@ -0,0 +1,121 @@
+using Microsoft.MixedReality.Toolkit.Tests.Services;
+using NUnit.Framework;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests.SpatialAwarenessSystem
+{
+    public class SpatialAwarenessSystemTests
+    {
+        private const string TestSpatialAwarenessSysteProfilePath = "Assets/MixedRealityToolkit.Tests/EditModeTests/Services/TestProfiles/TestMixedRealitySpatialAwarenessSystemProfile.asset";
+
+        [TearDown]
+        public void TearDown()
+        {
+            TestUtilities.ShutdownMixedRealityToolkit();
+        }
+
+        [Test]
+        public void TestGetSpatialAwarenessSystem()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+
+            // Retrieve Input System
+            IMixedRealitySpatialAwarenessSystem spatialAwarenessSystem = null;
+            MixedRealityServiceRegistry.TryGetService(out spatialAwarenessSystem);
+
+            // Tests
+            Assert.IsNotNull(spatialAwarenessSystem);
+        }
+
+        [Test]
+        public void TestSpatialAwarenessSystemDoesNotExist()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes();
+
+            // Check for Spatial Awareness System
+            Assert.IsFalse(MixedRealityToolkit.Instance.IsServiceRegistered<IMixedRealitySpatialAwarenessSystem>());
+        }
+
+        [Test]
+        public void TestSpatialAwarenessSystemExists()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+
+            // Check for Spatial Awareness System
+            Assert.IsTrue(MixedRealityToolkit.Instance.IsServiceRegistered<IMixedRealitySpatialAwarenessSystem>());
+        }
+
+        [Test]
+        public void TestEmptyDataProvider()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+
+            // Check for Input System
+            var spatialAwarenessSystem = MixedRealityToolkit.Instance.GetService<IMixedRealitySpatialAwarenessSystem>();
+            Assert.IsNotNull(spatialAwarenessSystem);
+
+            var dataProviderAccess = spatialAwarenessSystem as IMixedRealityDataProviderAccess;
+            Assert.IsNotNull(dataProviderAccess);
+            Assert.IsEmpty(dataProviderAccess.GetDataProviders());
+        }
+
+        [Test]
+        public void TestDataProviderRegisteration()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes();
+            MixedRealityToolkit.Instance.ActiveProfile.SpatialAwarenessSystemProfile = AssetDatabase.LoadAssetAtPath<MixedRealitySpatialAwarenessSystemProfile>(TestSpatialAwarenessSysteProfilePath);
+
+            var spatialAwarenessSystem = new MixedRealitySpatialAwarenessSystem(MixedRealityToolkit.Instance, MixedRealityToolkit.Instance.ActiveProfile.SpatialAwarenessSystemProfile);
+            Assert.IsNotNull(spatialAwarenessSystem);
```

Comment:
>Assert.IsNotNull(spatialAwarenessSystem); [](start = 12, length = 41)

Nit: Asserting null for creation of things via new isn't super interesting to verify (i.e. did I manage to new up memory?). If the constructor threw I think the test would have already failed out as well.